### PR TITLE
Add localization support

### PR DESCRIPTION
**Pull request by uncle-yura**
- Сreated a class for loading localizations.
- The mainwindow.py file has been edited to work with gettext.
- The startup.py file has been edited to work with gettext.
- Added Ukrainian and Russian translations for edited files.

### CODE REVIEW

File: awlsim/common/locale.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+#
+# AWL simulator - GUI
+#
+# Copyright 2020-2020 uncle-yura
```
Comment (mbuesch) [inline awlsim/common/locale.py]: I respect your desire to stay anonymous. But that actually is a real problem, in the (unlikely) event that there are copyright/licensing problems in the future. The only way to contact you would be through your github account. That's a real problem, especially for such a fundamental change, which effectively touches almost every file in the project.
For small projects I usually accept anonymous contributions, but awlsim is a different order of magnitude.

File: awlsim/common/locale.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+from awlsim.common.util import *
+import locale 
+import gettext
+import os
+import sys
+import getopt
+
```
Comment (mbuesch) [inline awlsim/common/locale.py]: This file should have an `__all__` list that only exports the `_` function.

File: awlsim/common/locale.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
```
Comment (mbuesch) [inline awlsim/common/locale.py]: For compatibility with different interpreters, every file in awlsim must do the following at the beginning of each module:

```
from __future__ import division, absolute_import, print_function, unicode_literals
#from awlsim.common.cython_support cimport * #@cy
from awlsim.common.compat import *
```

File: awlsim/common/locale.py (hunk 1 above)
Comment (uncle-yura) [reply awlsim/common/locale.py]: Ок, I added the email.

File: awlsim/common/locale.py (hunk 3 above)
Comment (uncle-yura) [reply awlsim/common/locale.py]: Added a header to the code.

File: awlsim/common/locale.py (hunk 2 above)
Comment (uncle-yura) [reply awlsim/common/locale.py]: Added export list.

File: awlsim/common/locale.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import locale 
+import gettext
+import os
+import sys
+import getopt
+
+class translate():
+	def __init__(self):
+		locale_dir = os.path.join(os.path.dirname(__file__),'..','..')
+		locale_dir = os.path.join( os.path.abspath(locale_dir) , "awlsim","locales" )
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/common/locale.py]: Yes, pkg_resources is a more elegant solution. Changed the code.

File: awlsim/common/locale.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		_locale, _encoding = locale.getdefaultlocale()
+
+		try:
+			(opts, args) = getopt.getopt(sys.argv[1:], "l:", [ "lang=", ])
+
+			for (o, v) in opts:
+				if o in ("-l", "--lang"):
+					_locale = str(v)		
+		except:
+			pass
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/common/locale.py]: Transferred parsing to a startup file.

File: awlsim/common/locale.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+				if o in ("-l", "--lang"):
+					_locale = str(v)		
+		except:
+			pass	
+		
+		try:
+			lang = gettext.translation("base", locale_dir, [_locale])
+			lang.install()
+			self._ = lang.gettext
+			printInfo(_("Using locale: {}").format(_locale))
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/common/locale.py]: Changed the code.

File: awlsim/common/locale.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+		except Exception as e:
+			printError("Unsupported locale ({}) setting or translation error ({})".format(_locale,e))
+			self._ = lambda s: s
+
+	def __call__(self,source):
+		return self._(source)
+
+_ = translate()
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/common/locale.py]: Changed the code.

File: awlsim/gui/mainwindow.py [hunk 8]
```diff
@@ -107,7 +109,7 @@ def __handleTopLevelChange(self, floating):
 		prefix = ""
 		if floating:
 			prefix = "%s - " % self.mainWidget.mainWindow.TITLE
-		self.setWindowTitle("%sCPU view" % prefix)
+		self.setWindowTitle(_("{}CPU view").format(prefix))
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/gui/mainwindow.py]: Moved the format function inside the translation processing. 
Now the lines should be of the form: 
`_("{} to {}", arg1,arg2)`
Inside the function, there are try-except, so if you do not succeed in formatting the translation, the source text will be displayed.

File: awlsim/locales/ru/LC_MESSAGES/base.po [hunk 9]
```diff
@@ -0,0 +1,502 @@
+msgid ""
```
[the remark this answers is not in the document]
Comment (uncle-yura) [reply awlsim/locales/ru/LC_MESSAGES/base.po]: I did not find a simple cross-platform way to compile mo files. So I added polib depending on the libraries and added the function to the setup script.

**Comment by mbuesch** [comment]
Thanks for your ongoing effort.
I didn't have time to fully check this for merge-ability, yet. But that shouldn't discourage you.

One minor development model thing though:
Please use the git rebase command. Please don't do merge-pulls. That results in merge-commits, which clutter up the history a lot.
And please also edit commits to include review findings instead of adding fixup commits. That also results in a cleaner history. The review process is still documented here in the pull request.
In case you don't know how to edit commits: git rebase --interactive helps to achieve that goal.
After a rebase a forced push is needed (-f).

**Pull request closed without merging**